MFVNeutralino/plugins/EventFilter.cc

```diff
@@ -6,119 +6,197 @@
+      leptons_pass = false;
+    }
+    else{
+      for (const pat::Electron& electron : *electrons) {
+        // if (electron_selector(electron) && electron.pt() > min_electron_pt)
+        //new electron selector : is cut based tight & iso < 0.10
+        reco::GsfTrackRef etk = electron.gsfTrack();
+        
+        if (etk.isNull()){
+          continue;
         }
-
-    const bool jets_pass = njets >= min_njets && ht >= min_ht;
-
-    if (debug) printf("MFVEventFilter: njets: %i  ht: %f pass? %i\n", njets, ht, jets_pass);
-
-    if (mode == Mode::jets_only) {
-        return jets_pass;
+        if (electron.pt() > min_electron_pt && abs(electron.eta()) < 2.4) {
+      
+          bool isTightEl = electron.electronID("cutBasedElectronID-Fall17-94X-V1-tight");
+          const auto pfIso = electron.pfIsolationVariables();
+          const float eA = electron_effective_areas.getEffectiveArea(fabs(electron.superCluster()->eta()));
+          const float iso = (pfIso.sumChargedHadronPt + std::max(0., pfIso.sumNeutralHadronEt + pfIso.sumPhotonEt - *rho*eA)) / electron.pt();
+          const bool passveto = electron.passConversionVeto();
+
+          if (isTightEl && passveto && iso < 0.10) {
+  	        ++nelectrons;
+          } 
+        }
+        leptons_pass = nmuons + nelectrons >= min_nleptons;
+        if (leptons_pass)
+          break;
+      }
     }
-    else if (mode == Mode::either && jets_pass)
-        return true;
-    else if (mode == Mode::HT_OR_bjets_OR_displaced_dijet || mode == Mode::bjets_OR_displaced_dijet_veto_HT || mode == Mode::bjets_OR_displaced_dijet || mode == Mode::displaced_dijet_veto_bjets)
-        return true;
-
-
-    edm::Handle<pat::MuonCollection> muons;
-    edm::Handle<pat::ElectronCollection> electrons;
-    event.getByToken(muons_token, muons);
-    event.getByToken(electrons_token, electrons);
-
-    int nmuons = 0, nelectrons = 0;
-
-    for (const pat::Muon& muon : *muons)
-        if (muon_selector(muon) && muon.pt() > min_muon_pt)
-            ++nmuons;
-
-    for (const pat::Electron& electron : *electrons)
-        if (electron_selector(electron) && electron.pt() > min_electron_pt)
-            ++nelectrons;
-
-    const bool leptons_pass = (nmuons + nelectrons >= min_nleptons);
-
-    if (debug) printf("MFVEventFilter: nmuons: %i nelectrons: %i pass? %i\n", nmuons, nelectrons, leptons_pass);
-
-    return leptons_pass;
-}
-
+  }
+    
+  if (debug) printf("MFVEventFilter: nmuons: %i nelectrons: %i pass? %i\n", nmuons, nelectrons, leptons_pass);
+
+  if (mode == Mode::jets_only)
+    return jets_pass;
+  else if (mode == Mode::either && jets_pass)
+    return true;
+  else if (mode == Mode::HT_OR_bjets_OR_displaced_dijet || mode == Mode::bjets_OR_displaced_dijet_veto_HT)
+    return true;
+  else if (mode == Mode::MET_only)
+    return true;
+  else if (mode == Mode::muons_only || mode == Mode::electrons_only_veto_muons)
+    return leptons_pass && jets_pass;
+  else return true; // catch-all
```

Comment:
Should the catch all just throw an error instead? I may be forgetting if e.g. signals would reach this point, or if they never do...